### PR TITLE
Split vc

### PR DESCRIPTION
**Pull request by khrystyna-sb**

### CODE REVIEW

File: Countries/CountryTableViewCell.swift [hunk 1]
```diff
@@ -81,15 +81,15 @@ class CountryTableViewCell: UITableViewCell {
...
+    private func setupLayoutConstraints() {
         let imageSideSize = contentView.frame.size.height - LayoutConstants.indent
         
         NSLayoutConstraint.activate([
             
             flagImageView.heightAnchor.constraint(equalToConstant: LayoutConstants.imageHeight),
             flagImageView.widthAnchor.constraint(equalToConstant: LayoutConstants.imageWidth),
             flagImageView.topAnchor.constraint(equalTo: contentView.topAnchor, constant: LayoutConstants.indent),
-            flagImageView.trailingAnchor.constraint(equalTo: stackView.leadingAnchor),
+//            flagImageView.trailingAnchor.constraint(equalTo: stackView.leadingAnchor),
```
Comment (Globalika) [inline Countries/CountryTableViewCell.swift]: do you need this comment line?

File: Countries/SceneDelegate.swift [hunk 2]
```diff
@@ -8,22 +8,34 @@
...
+        let detailNavigationController = UINavigationController(rootViewController: detailController)
+
+        let splitViewController =  UISplitViewController()
+        splitViewController.viewControllers = [masterNavigationController, detailNavigationController]
+        
+        splitViewController.delegate = self
+        
+        self.window = UIWindow(windowScene: windowScene)
+        self.window!.rootViewController = splitViewController
+        self.window!.makeKeyAndVisible()
```
Comment (Globalika) [inline Countries/SceneDelegate.swift]: maybe it should be optional chaining?

File: Countries/SceneDelegate.swift [hunk 3]
```diff
@@ -46,7 +58,13 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
...
     }
+    
+    
+}
 
-
+extension SceneDelegate: UISplitViewControllerDelegate {
+    func splitViewController(_ splitViewController: UISplitViewController, collapseSecondary secondaryViewController: UIViewController, onto primaryViewController: UIViewController) -> Bool {
+         return true
+     }
```
Comment (Globalika) [inline Countries/SceneDelegate.swift]: looks like extra space

**Review by Globalika — CHANGES_REQUESTED** [review]
I left you some comments

**Review by Globalika — APPROVED** [review]

File: Countries/ListViewController.swift [hunk 4]
```diff
@@ -11,50 +11,48 @@ enum TableConstants {
     static let heightForRow: CGFloat = 100.0
 }
 
-class ListViewController: UIViewController {
+class ListViewController: UITableViewController {
```
Comment (apolwork) [inline Countries/ListViewController.swift:14]: :)

FYI: It was possible to fix your issue with table view without migrating to UITableViewController

File: Countries/SceneDelegate.swift [hunk 5]
```diff
@@ -8,45 +8,63 @@
 import UIKit
 
 class SceneDelegate: UIResponder, UIWindowSceneDelegate {
-
+    
     var window: UIWindow?
-
-
+    
+
```
Comment (apolwork) [inline Countries/SceneDelegate.swift]: Empty spaces....

File: Countries/SceneDelegate.swift [hunk 6]
```diff
@@ -8,45 +8,63 @@
...
     }
-
+    
     func sceneDidDisconnect(_ scene: UIScene) {
-        // Called as the scene is being released by the system.
-        // This occurs shortly after the scene enters the background, or when its session is discarded.
-        // Release any resources associated with this scene that can be re-created the next time the scene connects.
-        // The scene may re-connect later, as its session was not necessarily discarded (see `application:didDiscardSceneSessions` instead).
+        
     }
```
Comment (apolwork) [inline Countries/SceneDelegate.swift]: Do we need this empty method?

File: Countries/SceneDelegate.swift [hunk 7]
```diff
@@ -8,45 +8,63 @@
...
-
+    
     func sceneDidBecomeActive(_ scene: UIScene) {
         // Called when the scene has moved from an inactive state to an active state.
         // Use this method to restart any tasks that were paused (or not yet started) when the scene was inactive.
     }
```
Comment (apolwork) [inline Countries/SceneDelegate.swift]: saa

File: Countries/SceneDelegate.swift [hunk 8]
```diff
@@ -8,45 +8,63 @@
...
         // Called as the scene transitions from the background to the foreground.
         // Use this method to undo the changes made on entering the background.
     }
-
+    
     func sceneDidEnterBackground(_ scene: UIScene) {
         // Called as the scene transitions from the foreground to the background.
         // Use this method to save data, release shared resources, and store enough scene-specific state information
         // to restore the scene back to its current state.
     }
```
Comment (apolwork) [inline Countries/SceneDelegate.swift]: saa

File: Countries/SceneDelegate.swift [hunk 9]
```diff
@@ -8,45 +8,63 @@
...
+    
+
```
Comment (apolwork) [inline Countries/SceneDelegate.swift]: Empty spaces )

File: Countries/SceneDelegate.swift [hunk 10]
```diff
@@ -8,45 +8,63 @@
...
+        let splitViewController =  UISplitViewController()
+        splitViewController.viewControllers = [masterNavigationController, detailNavigationController]
+        
+        splitViewController.delegate = self
+        
+        self.window = UIWindow(windowScene: windowScene)
+        self.window?.rootViewController = splitViewController
+        self.window?.makeKeyAndVisible()
+        
+        // scene bounds doesn't reload
```
Comment (apolwork) [inline Countries/SceneDelegate.swift]: What this comment mean?

**Review by apolwork — CHANGES_REQUESTED** [review]

File: Countries/ListViewController.swift (hunk 4 above)
Comment (khrystyna-sb) [reply Countries/ListViewController.swift:14]: Yes, but there is no reason to not using TableViewController in this project) so, I've decided to migrate to it

File: Countries/SceneDelegate.swift (hunk 6 above)
Comment (khrystyna-sb) [reply Countries/SceneDelegate.swift]: Not now.. I deleted them and will add back when/if I need :)

File: Countries/SceneDelegate.swift (hunk 10 above)
Comment (khrystyna-sb) [reply Countries/SceneDelegate.swift]: Saory, it's irrelevant already

**Review by apolwork — APPROVED** [review]

**Pull request merged**